src/OpenLoco/src/GameCommands/RaiseLowerLand.cpp

```diff
@@ -0,0 +1,485 @@
+
+            auto height = el.baseZ() - targetBaseZ;
+            if (height < 0)
+            {
+                continue;
+            }
+
+            if (el.type() == ElementType::track)
+            {
+                auto* trackEl = el.as<TrackElement>();
+                if (trackEl != nullptr)
+                {
+                    if (trackEl->hasBridge())
+                    {
+                        auto* bridgeObj = ObjectManager::get<BridgeObject>(trackEl->bridge());
+                        if (height > bridgeObj->maxHeight)
+                        {
+                            setErrorText(StringIds::bridge_already_at_maximum_height);
+                            return FAILURE;
+                        }
+                    }
+
+                    auto* trackObj = ObjectManager::get<TrackObject>(trackEl->trackObjectId());
+                    FormatArguments args{};
+                    args.push(trackObj->name);
+                    setErrorText(StringIds::stringid_requires_a_bridge);
+                    return FAILURE;
+                }
+            }
+            else if (el.type() == ElementType::road)
+            {
+                auto* roadEl = el.as<RoadElement>();
+                if (roadEl != nullptr)
+                {
+                    if (roadEl->hasBridge())
+                    {
+                        auto* bridgeObj = ObjectManager::get<BridgeObject>(roadEl->bridge());
+                        if (height > bridgeObj->maxHeight)
+                        {
+                            setErrorText(StringIds::bridge_already_at_maximum_height);
+                            return FAILURE;
+                        }
+                    }
+
+                    auto* roadObj = ObjectManager::get<RoadObject>(roadEl->roadObjectId());
+                    FormatArguments args{};
+                    args.push(roadObj->name);
+                    setErrorText(StringIds::stringid_requires_a_bridge);
+                    return FAILURE;
+                }
+            }
+        }
+
+        if (!(flags & Flags::apply))
+        {
+            return 0;
+        }
+
+        surface = tileIt.surface();
+        if (isEditorMode())
+        {
+            // Remove 'recently cleared' status from terrain in editor
+            surface->setTerrain(surface->terrain());
+        }
+
+        surface->setBaseZ(targetBaseZ);
+        surface->setClearZ(targetBaseZ);
+        surface->setSlope(slopeFlags);
+
+        landObj = ObjectManager::get<LandObject>(surface->terrain());
+        if (landObj->hasFlags(LandObjectFlags::unk1) && isEditorMode())
+        {
+            surface->setTerrain(landObj->var_07);
+        }
+
+        if (surface->water() <= targetBaseZ)
+        {
+            surface->setWater(0);
+        }
+
+        TileManager::mapInvalidateTileFull(pos);
+        return totalCost;
+    }
+
+    // 0x00463702, 0x0046370B
+    // NB: resetF00170 is used for 0x00463702; it is omitted by 0x0046370B
```

Comment:
resetF00170 is the ` std::set<World::Pos3, LessThanPos3> removedBuildings{}` so by the sounds of it `0x0046370B` should be passing removedBuildings into this function and `0x00463702` shouldn't.

Reply:
I think you're right, the main game command functions (i.e. the ones that currently get registers passed) should probably initialise these lists instead. Is that what you had in mind as well?

Reply:
Yeah